tachyon/math/matrix/matrix_utils.h

```diff
@@ -72,6 +72,35 @@ std::vector<PackedField*> PackRowHorizontally(
   });
 }
 
+// Creates a vector of packed fields for a given matrix row. If the length
+// of the row is not a multiple of |PackedField::N|, the last |PackedField|
+// element populates leftover values with |F::Zero()|.
+template <typename Expr, int BlockRows, int BlockCols, bool InnerPanel,
+          typename PackedField =
+              typename PackedFieldTraits<typename Expr::Scalar>::PackedField>
+std::vector<PackedField> PackRowHorizontallyPadded(
+    const Eigen::Block<Expr, BlockRows, BlockCols, InnerPanel>& matrix_row) {
+  size_t cols = static_cast<size_t>(matrix_row.cols());
+  using F = typename FiniteFieldTraits<PackedField>::PrimeField;
```

Comment:
The type "F" never be used in this function.

Reply:
It is used in the `F::Zero();` if else statement of the second half of the function